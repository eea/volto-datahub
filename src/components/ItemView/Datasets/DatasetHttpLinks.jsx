import { Icon, List } from 'semantic-ui-react';
import { isInternalURL } from '@eeacms/volto-datahub/utils';
import { SVGIcon } from '../utils.js';

import lockSVG from 'remixicon/icons/System/lock-line.svg';

export default function DatasetHttpLinks({ dataset }) {
  return dataset['WWW:LINK-1.0-http--link'] || dataset['DOI'] ? (
    <>
      <h5>Links:</h5>
      <List divided relaxed>
        {(dataset['WWW:LINK-1.0-http--link'] || []).map((item, i) => {
          return (
            <List.Item key={i}>
              <List.Content>
                <div className="dataset-item">
                  <Icon className="linkify" />
                  <a
                    href={item.url}
                    className="item-link"
                    target="_blank"
                    rel="noreferrer"
                  >
                    <span>{item.name || item.url}</span>
                  </a>
                  {isInternalURL(item.url) ? (
                    <SVGIcon name={lockSVG} size="20px" />
                  ) : null}
                </div>
              </List.Content>
              {item.description ? (
                <span className="item-description">{item.description}</span>
              ) : null}
            </List.Item>
          );
        })}
        {(dataset['DOI'] || []).map((item, i) => {
          return (
            <List.Item key={i}>
              <List.Content>
                <div className="dataset-item">
                  <Icon className="info circle" />
                  <a
                    href={item.url}
                    className="item-link"
                    target="_blank"
                    rel="noreferrer"
                  >
                    <span>{item.name || item.url}</span>
                  </a>
                  {isInternalURL(item.url) ? (
                    <SVGIcon name={lockSVG} size="20px" />
                  ) : null}
                </div>
              </List.Content>
              {item.description ? (
                <span className="item-description">{item.description}</span>
              ) : null}
            </List.Item>
          );
        })}
      </List>
    </>
  ) : null;
}
