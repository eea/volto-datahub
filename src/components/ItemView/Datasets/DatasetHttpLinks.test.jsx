import React from 'react';
import renderer from 'react-test-renderer';

import DatasetHttpLinks from './DatasetHttpLinks';

jest.mock('@eeacms/search', () => ({
  runRequest: jest.fn(),
}));

const dataset = {
  'WWW:LINK-1.0-http--link': [
    {
      id: 1,
      url: 'https://example.org/1',
      protocol: 'https',
      name: 'First',
      description: 'Something here',
    },
  ],
  DOI: [
    {
      id: 1,
      url: 'https://example.org/1',
      protocol: 'https',
      name: 'First',
      description: 'Something here',
    },
  ],
};

describe('DatasetHttpLinks', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<DatasetHttpLinks dataset={dataset} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
