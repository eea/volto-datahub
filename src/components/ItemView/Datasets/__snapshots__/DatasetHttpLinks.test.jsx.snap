// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetHttpLinks renders correctly 1`] = `
Array [
  <h5>
    Links:
  </h5>,
  <div
    className="ui divided relaxed list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="content"
      >
        <div
          className="dataset-item"
        >
          <i
            aria-hidden="true"
            className="icon linkify"
            onClick={[Function]}
          />
          <a
            className="item-link"
            href="https://example.org/1"
            target="_blank"
          >
            <span>
              First
            </span>
          </a>
        </div>
      </div>
      <span
        className="item-description"
      >
        Something here
      </span>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="content"
      >
        <div
          className="dataset-item"
        >
          <i
            aria-hidden="true"
            className="icon info circle"
            onClick={[Function]}
          />
          <a
            className="item-link"
            href="https://example.org/1"
            target="_blank"
          >
            <span>
              First
            </span>
          </a>
        </div>
      </div>
      <span
        className="item-description"
      >
        Something here
      </span>
    </div>
  </div>,
]
`;
